src/main/java/doob/model/SurvivalLevel.java

```diff
@@ -0,0 +1,109 @@
+package doob.model;
+
+import doob.util.BoundsTuple;
+
+import java.util.Random;
+
+/**
+ * A survival level.
+ * Created by hidde on 10/21/15.
+ */
+public class SurvivalLevel extends Level {
+
+    private int nextWaveHP = 0;
+    private static final int DIFFICULTY_TIME = 1000;
+    private static final int SMALLEST_BALL_SIZE = 16;
+    private static final int NUMBER_POSSIBLE_BALLS = 5;
+    private BoundsTuple bounds;
+
+    /**
+     * Init new level with size bounds.
+     *
+     * @param bounds width and height
+     */
+    public SurvivalLevel(BoundsTuple bounds) {
+        super(bounds);
+        this.bounds = bounds;
+    }
+
+    @Override
+    public void update() {
+        super.update();
+        double cTime = getCurrentTime();
+        if (cTime > DIFFICULTY_TIME) {
+            setCurrentTime(cTime - DIFFICULTY_TIME);
+            nextWaveHP++;
+        }
+        if (totalBallHitpoints() <= nextWaveHP) {
+            spawnBalls(System.currentTimeMillis());
+        }
+    }
+
+    /**
+     * Adds one to time.
+     * @return one
+     */
+    @Override
+    protected int getTimeMutation() {
```

Comment:
Ah, that is the function that gets 1 or -1, depending on whether it is called from Level or SurvivalLevel. This way, we don't have to implement update() twice, just override the getTimeMutation method.